### ISSUE TITLE
git for windows 2.7.0 bash CLI pauses for input then prints "it's a number" or "it's not a number" before the branch spec

### ISSUE THREAD

**Issue by dalpert-korewireless** [git-for-windows/git#609]
Given I install Git for Windows 2.7.0
And open a bash shell
And change into a git working directory
When I press `<enter>` at a command prompt
Then I expect to see the next command prompt right away.

Instead, I was seeing the command shell pause to read input from the command line.

Then, based on whether that input was parseable to a number or not, the prompt would
print out "it's a number" or "it's not a number" as part of the decoration configured by PS1
out of the box (see example session below)

````
DAlpert@machine-name MINGW64 /d/dev/git/repo-nameit's not a number (patch-8.21.3)
$ <enter>
<enter>
DAlpert@machine-name MINGW64 /d/dev/git/repo-nameit's not a number (patch-8.21.3)
$ <enter>
4<enter>
DAlpert@machine-name MINGW64 /d/dev/git/repo-nameit's a number (patch-8.21.3)
$
````

This took a while to track down as initially I didn't realize it was waiting on input but
rather wondered if the newest version of Git for Windows shell had introduced an 
interpreter-like requirement to press `<enter>` twice after every command.

This is particularly confusing when entering a command like a directory listing and
wondering why it was not returning me to the command prompt until after I pressed
<enter> again.

Using a manual form of `git bisect` I was able to trace the source of this behavior 
to [this line in `git-prompt.sh`](https://github.com/git-for-windows/git/blob/v2.7.0/contrib/completion/git-prompt.sh#L433)

For reference, the contents of my ./.git/HEAD file in this case are:
````
ref: refs/heads/patch-8.21.3
````

**Comment by dalpert-korewireless** [comment]
I have found that replacing [Line 433 of `git-prompt.sh`](https://github.com/git-for-windows/git/blob/v2.7.0/contrib/completion/git-prompt.sh#L433) with the following eliminates the problem for me:
````
head="$(cat $g/HEAD)"
````

**Comment by dalpert-korewireless** [comment]
drat; looks like I was having the issue with the [v2.7.0.windows.1](https://github.com/git-for-windows/git/releases/download/v2.7.0.windows.1/Git-2.7.0-64-bit.exe) build from https://git-scm.com/download/win 

As such my links should go against the v2.7.0.windows.1 tag instead of v2.7.0:
- [`git-prompt.sh` - Line 433](https://github.com/git-for-windows/git/blob/v2.7.0.windows.1/contrib/completion/git-prompt.sh#L433)

And PR #610 into `master` may be targeting the wrong branch...

**Comment by rimrul** [comment]
Thank you for investigating your issue this deeply and sharing your solution. 

>Using a manual form of git bisect I was able to trace the source of this behavior
to this line in git-prompt.sh

>I have found that replacing Line 433 of git-prompt.sh with the following eliminates the problem for me: head="$(cat $g/HEAD)"

I'm not sure, that [`__git_eread`](https://github.com/dalpert-korewireless/git/blob/patch-1/contrib/completion/git-prompt.sh#L280-L285) is the real cause here. I had a look at it, and it doesn't seem to check if the last input is a number. Could you have a look at your `~/.bashrc` and `~/.bash_profile`?

**Comment by dalpert-korewireless** [comment]
@rimrul thank you for consideration of this issue.  while preparing a gist of my `~/.bashrc` (and related scripts) I realized that one of the files my `~/.bashrc` sources on startup (one which I had not looked at in some time) had redefined [`test`](https://github.com/dalpert-korewireless/git/blob/patch-1/contrib/completion/git-prompt.sh#L284).  Lo and behold, there in my own forgotten startup script was a test of input which was experimenting with bash conditionals by printing, you guessed it, "it's a number" or "it's not a number".

A little knowledge is a dangerous thing. I did not know that `test` was a pre-existing function in bash.

Commenting out the (re)defined `test` method in my startup scripts I was able to revert Line 433 of git-prompt.sh back to what is shipped with v2.7.0.windows.1 and it works as expected without issue.
